core/state/state_object.go

```diff
@@ -370,8 +355,8 @@ func (s *stateObject) commitTrie(db Database) (*trie.NodeSet, error) {
 	if err != nil {
 		return nil, err
 	}
-	if s.dbErr != nil {
-		return nil, s.dbErr
+	if s.db.dbErr != nil {
```

Comment:
This check is a bit weird. `commitTrie` is only called from `StateDB.Commit`, which already checks for `s.db.dbErr` in the beginning. I think this check should be removed.